pkgs/applications/science/computer-architecture/timeloop/default.nix

```
@@ -0,0 +1,97 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, scons
+, libconfig
+, boost
+, libyaml
+, libyamlcpp
+, ncurses
+, gpm
+, enableAccelergy ? true
+, enableISL ? false
+, accelergy
+}:
+
+stdenv.mkDerivation rec {
+  pname = "timeloop";
+  version = "unstable-2022-10-28";
+
+  src = fetchFromGitHub {
+    owner = "NVlabs";
+    repo = "timeloop";
+    rev = "83088f2b9b01581685cfe856dd9d84e46575cfdb";
+    hash = "sha256-X/5lvOYSPJTKcdgMEKVQbd34n4INf7P+lTl98yv82fs=";
+  };
+
+  nativeBuildInputs = [ scons ];
+
+  buildInputs = [
+    libconfig
+    boost
+    libyaml
+    libyamlcpp
+    ncurses
+    accelergy
+   ] ++ lib.optionals stdenv.isLinux [ gpm ];
+
+  preConfigure = ''
+     cp -r ./pat-public/src/pat ./src/pat
+   '';
+
+  enableParallelBuilding = true;
+
+  #link-time optimization fails on darwin
+  #see https://github.com/NixOS/nixpkgs/issues/19098
+  NIX_CFLAGS_COMPILE = lib.optional stdenv.isDarwin "-fno-lto";
+
+  patchPhase = ''
+    #use nix ar/ranlib
+    substituteInPlace ./SConstruct --replace "env.Replace(AR = \"gcc-ar\")" ""
+    substituteInPlace ./SConstruct --replace "env.Replace(RANLIB = \"gcc-ranlib\")" ""
+    '' + lib.optionalString stdenv.isDarwin ''
+    #prevent clang from dying on errors that gcc is fine with
+    substituteInPlace ./src/SConscript --replace "-Werror" "-Wno-inconsistent-missing-override"
+
+    #disable LTO on macos
+    substituteInPlace ./src/SConscript --replace ", '-flto'" ""
+
+    #static builds on mac fail as no static libcrt is provided by apple
+    #see https://stackoverflow.com/questions/3801011/ld-library-not-found-for-lcrt0-o-on-osx-10-6-with-gcc-clang-static-flag
+    substituteInPlace ./src/SConscript --replace "'-static-libgcc', " ""
+    substituteInPlace ./src/SConscript --replace "'-static-libstdc++', " ""
+    substituteInPlace ./src/SConscript --replace "'-Wl,--whole-archive', '-static', " ""
+    substituteInPlace ./src/SConscript --replace ", '-Wl,--no-whole-archive'" ""
+
+    #remove hardcoding of gcc
+    sed -i '40i env.Replace(CC = "${stdenv.cc.targetPrefix}cc")' ./SConstruct
+    sed -i '40i env.Replace(CXX = "${stdenv.cc.targetPrefix}c++")' ./SConstruct
+
+    #gpm doesn't exist on darwin
+    substituteInPlace ./src/SConscript --replace ", 'gpm'" ""
+   '';
+
+  sconsFlags =
+    # will fail on clang/darwin on link without --static due to undefined extern
+    # however, will fail with static on linux as nixpkgs deps aren't static
+    (lib.optional stdenv.isDarwin "--static")
+    ++ (lib.optional enableAccelergy "--accelergy")
+    ++ (lib.optional enableISL "--with-isl");
+
+
+  installPhase = ''
+    cp -r ./bin $out
+    cp -r ./lib $out
+    cp -r ./problem-shapes $out
+    cp -r ./doc $out
+    cp -r ./configs $out
```

```suggestion
    cp -r ./bin ./lib ./problem-shapes ./doc ./configs $out
```
problem-shapes and configs should go into a subdirectory. doc should go under share